libvast/src/http_api.cpp

```diff
@@ -8,12 +8,34 @@
 
 #include <vast/concept/parseable/to.hpp>
 #include <vast/concept/parseable/vast/data.hpp>
+#include <vast/concept/printable/vast/json.hpp>
 #include <vast/http_api.hpp>
 
 #include <fmt/format.h>
 
 #include <simdjson.h>
 
+namespace {
+
+// Validates that the argument is a valid JSON object.
+auto validate_json(const std::string& json) -> bool {
+  auto parser = simdjson::dom::parser{};
+  auto padded = simdjson::padded_string{json};
+  auto obj = simdjson::dom::object{};
+  return parser.parse(json).get(obj) == simdjson::SUCCESS;
```

Comment:
This does not work, because `simdjson` does lazy validation.

Reply:
Also for the `dom` parser? I thought it was only for the `ondemand` parser.

Reply:
Sorry, you are correct :+1: I thought it would be `ondemand`, but you are using `dom`.